homework/src/index.js

```diff
@@ -1,47 +1,171 @@
+        createAndAppend('td', forkLine, {
+            html: 'Forks :',
+            class: 'label'
+        });
+        const forkValue = createAndAppend('td', forkLine);
+        const updateLine = createAndAppend('tr', table);
+        createAndAppend('td', updateLine, {
+            html: 'Updated :',
+            class: 'label'
+        });
+        const updateValue = createAndAppend('td', updateLine);
+        repoNameLink.innerText = repository.name;
+        repoNameLink.setAttribute('href', repository.html_url);
+        if (repository.description === null) {
+            descriptionLabel.innerHTML = '';
+            descriptionValue.innerText = '';
+        } else {
+            descriptionLabel.innerText = "Description :";
+            descriptionValue.innerText = repository.description;
+        }
+        forkValue.innerText = repository.forks;
+        const updateRepo = new Date(repository.updated_at);
+        updateValue.innerText = updateRepo.toLocaleString();
+    }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+    function fetchAndShowContributors(contributorsUrl) {
+        fetchJSON(contributorsUrl)
+            .then((contributors) => {
+                contributors.forEach(contributor => {
+                    const li = createAndAppend('li', ul);
+                    createAndAppend('img', li, {
+                        src: contributor.avatar_url
+                    });
+                    createAndAppend('a', li, {
+                        html: contributor.login,
+                        href: contributor.html_url,
+                        target: '_blank'
+                    });
+                    createAndAppend('div', li, {
+                        html: contributor.contributions,
+                        class: 'contributionNum'
+                    });
+                });
+            })
+            .catch((err) => console.log(err));
+    }
 
-  window.onload = () => main(HYF_REPOS_URL);
+    function main(url) {
+
+        fetchJSON(url)
+            .then((repositories) => {
+                repositories.sort((a, b) => a.name.localeCompare(b.name));
+                repositories.forEach((repository, i) => {
+                    createAndAppend('option', select, {
+                        html: repository.name,
+                        value: i
+                    });
+                });
+                ShowRepository(repositories[0]);
+                const contributorsUrl = repositories[0].contributors_url;
+                fetchAndShowContributors(contributorsUrl);
+
+                select.addEventListener('change', (e) => {
+                    const index = e.target.value;
+                    ShowRepository(repositories[index]);
+                    ul.innerHTML = '';
+                    const contributorsUrl = repositories[index].contributors_url;
+                    fetchAndShowContributors(contributorsUrl);
```

Comment:
You will notice that there is a slight delay between showing the repo info and the contributors info. Time-wise it looks like this.

[show repo][clear contributors container][fetch contributors].......[show contributors]

You can improve on the user experience by postponing the (re-)rendering until the contributor data has been returned:

[fetch contributors].......[clear contributors container][render contributors][show repo]

Now, while the contributors are being fetched the screen remains unchanged.

You can simply achieve this by moving the clearing out of the contributors `ul` to the `fetchAndShowContributors()` function, like this:

```js
function fetchAndShowContributors(ul, contributorsUrl) {
  fetchJSON(contributorsUrl)
    .then((contributors) => {
      ul.innerHTML = '';
      //...
```

and moving the `showRepositories()` function a bit:

```js
fetchAndShowContributors(ul, contributorsUrl);
showRepository(repositoriesInfoSec, repositories[index]);
```

Reply:
Thank you for your helpful feedback @remarcmij  👍